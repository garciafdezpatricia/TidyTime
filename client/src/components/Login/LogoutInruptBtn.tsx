// Copyright 2024 Patricia García Fernández.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { useInruptHandler } from "@/pages/api/inrupt";
import { useTranslation } from 'react-i18next';


export default function LogoutInrupt() {
    const { t } = useTranslation();
    const { logoutInrupt } = useInruptHandler();

    return (
        <button data-testid='inrupt-logout-button' className="logout-inrupt-button" onClick={logoutInrupt}>
            {t('home.logoutButton')}
        </button>
    )
}