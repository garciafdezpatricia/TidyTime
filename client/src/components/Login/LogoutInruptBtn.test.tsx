import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import LogoutInruptBtn from '@/src/components/Login/LogoutInruptBtn';

describe('LogoutInruptBtn', () => {
    it('renders without crashing', () => {
      const { container } = render(<LogoutInruptBtn/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<LogoutInruptBtn/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });