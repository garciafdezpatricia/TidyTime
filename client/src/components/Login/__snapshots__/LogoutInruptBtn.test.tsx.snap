// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutInruptBtn matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="logout-inrupt-button"
    data-testid="inrupt-logout-button"
  >
    home.logoutButton
  </button>
</DocumentFragment>
`;
